pkg/provider/utils.go

```diff
@@ -0,0 +1,28 @@
+package provider
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/EnterpriseDB/terraform-provider-biganimal/pkg/api"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+)
+
+func FromBigAnimalErr(err error) diag.Diagnostics {
```

Comment:
we should make this lower case, so that we don't export it 
keep the interface small

Reply:
Ah, good catch! Done.